From 4e16ad9ec93d977b053f9c1fb61ed794333f1013 Mon Sep 17 00:00:00 2001
From: Xulin Sun <xulin.sun@windriver.com>
Date: Mon, 11 Dec 2017 11:27:54 +0800
Subject: [PATCH] arm64: kexec: correct variable uninitialized warning

To fix below building warning:

kernel/kexec_core.c: In function '__crash_kexec':
arch/arm64/include/asm/kexec.h:65:3: warning: 'tmp1' may be used uninitialized in this function [-Wmaybe-uninitialized]
    __asm__ __volatile__ (
    ^~~~~~~
arch/arm64/include/asm/kexec.h:63:7: note: 'tmp1' was declared here
    u64 tmp1, tmp2;
        ^~~~
arch/arm64/include/asm/kexec.h:65:3: warning: 'tmp2' may be used uninitialized in this function [-Wmaybe-uninitialized]
    __asm__ __volatile__ (
    ^~~~~~~
arch/arm64/include/asm/kexec.h:63:13: note: 'tmp2' was declared here
    u64 tmp1, tmp2;
              ^~~~

Signed-off-by: Xulin Sun <xulin.sun@windriver.com>
---
 arch/arm64/include/asm/kexec.h |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/arch/arm64/include/asm/kexec.h b/arch/arm64/include/asm/kexec.h
index c6003c7..adecc15 100644
--- a/arch/arm64/include/asm/kexec.h
+++ b/arch/arm64/include/asm/kexec.h
@@ -93,7 +93,7 @@ static inline void crash_setup_regs(struct pt_regs *newregs,
 			"adr	 %1, 1f\n"
 		"1:\n"
 			"stp	 %1, %0,   [%2, #16 * 16]\n"
-			: "+r" (tmp1), "+r" (tmp2)
+			: "=&r" (tmp1), "=&r" (tmp2)
 			: "r" (newregs)
 			: "memory"
 		);
-- 
1.7.5.4

